REVIEW_GUIDELINES.md

```diff
@@ -83,3 +83,22 @@ image:
 ## Compatibility
 
 We officially support compatibility with the current and the previous minor version of Kubernetes. Generated resources should use the latest possible API versions compatible with these versions. For extended backwards compatibility conditional logic based on capalilities may be used (see [built-in objects](https://github.com/kubernetes/helm/blob/master/docs/chart_template_guide/builtin_objects.md)).
+
+## Kubernetes Native Workloads.
+
+While reviewing Charts that contains workloads such as [Deployment](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/), [Statefulset](https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/), [Daemonset](https://kubernetes.io/docs/concepts/workloads/controllers/daemonset/) and [Jobs](https://kubernetes.io/docs/concepts/workloads/controllers/jobs-run-to-completion/) below points should be considered.  These are to seen as best practises rather than strict enforcement.
+  
+1. Any workload that are stateless and long running (servers) in nature are to be created as deployments.  Deployments in turn creates replica sets.
+2. Unless there is a compelling reason replica sets or replica controllers are are to be avoided as workload types. 
+3. Workloads that are stateful in nature such as Databases, Key Value Store, Messages Queues , In-memory caches are to be created as statefulsets
+4. It is recommended that Deployments and Statefulsets configure their workloads with [Pod Disruption Budget](https://kubernetes.io/docs/concepts/workloads/pods/disruptions/) for high availability.
+5. For workloads such as databases, KV store, etc., that replicates data it is recommended to configure interpod anti-affinity.
+6. It is recommended to have default workload update strategy configured that is suitable for this chart
+7. Batch workloads are to be created using Jobs. 
+8. It is best to always create workload with latest supported [api version](    https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.9/#statefulset-v1-apps) as older version are either depcicated or soon to be depreciated. 
```

Comment:
The latest is only supported in 1.9. `apps/v1` isn't supporting in 1.8. Because of these we should use the last api version. Many providers don't even provide 1.9 yet (like GKE).